src/kernel/src/process/rlimit.rs

```diff
@@ -1,4 +1,72 @@
-use std::io::Error;
+use std::{io::Error, ops::Index};
+use thiserror::Error;
+
+//TODO: add remaining limits
+#[derive(Debug, Clone, Copy)]
+pub enum ResourceType {
+    Cpu = 0,
+    Fsize = 1,
+    Data = 2,
+}
+
+#[cfg(target_os = "linux")]
+type RawResourceType = libc::__rlimit_resource_t;
+
+#[cfg(target_os = "macos")]
+type RawResourceType = libc::c_int;
+
+impl ResourceType {
+    #[cfg(unix)]
+    pub fn into_host(self) -> RawResourceType {
+        match self {
+            Self::Cpu => libc::RLIMIT_CPU,
+            Self::Fsize => libc::RLIMIT_FSIZE,
+            Self::Data => libc::RLIMIT_DATA,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub(super) struct Limits {
+    inner: [ResourceLimit; Self::NLIMITS],
+}
+
+impl Limits {
+    pub const NLIMITS: usize = 3;
```

Comment:
This one is quite dangerous because if it is out of sync with `ResourceType` we will hit undefined behavior in the `Index<ResourceType>` implementation.

Reply:
@SuchAFuriousDeath seems like you are missed this one.

Reply:
I changed it to get(t as usize).unwrap, so no UB anymore

Reply:
Do you know is it possible to get the maximum value of the enum type? If that possible it will be much more elegance than this one.

Reply:
I looked, but it's nightly-only :(
https://doc.rust-lang.org/std/mem/fn.variant_count.html

Reply:
I got an idea but you don't need to implement it if you are lazy. The idea is create a derive macro to find out the maximum value of the variants then generate an implementation something like:

```rust
impl ResourceType {
    pub const MAX: usize = 2;
}
```

Reply:
Unfortunately, I am lazy :( I am playing around with slint though.

Reply:
Okay no problem. We will go with the current solution for now.